server/src/main/java/io/crate/replication/logical/LogicalReplicationSettings.java

```diff
@@ -56,6 +58,15 @@
         Setting.Property.NodeScope
     );
 
+    public static final Setting<ByteSizeValue> RECOVERY_CHUNK_SIZE =
```

Comment:
Exposed and moved to the LR settings to keep them all together. Not sure whether I need to add `addSettingsUpdateConsumer` method - if I understand correctly, setting is [updated](https://github.com/crate/crate/blob/2a300d4fb0776accf1e17e3c35f3b032ca2cc2cf/server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java#L192) anyway, it's rather needed when we inject `LogicalReplicationSettings` and use getter but for simplicity I kept  the only direct usage `RECOVERY_CHUNK_SIZE.get(settings) ` in the `LogicalReplicationRepository`

Reply:
`RECOVERY_CHUNK_SIZE.get(settings)` uses the injected settings which will not change until a node restart. Thus to have it dynamically changeable, we'd have to use a settings consumer which will apply an updated setting out of the cluster state.

Reply:
I got it wrong, thx for the clarification, will fix

Reply:
Drafted in the last commit, currently blocked with passing LogicalReplicationSettings https://github.com/crate/crate/pull/12142/files#diff-d01b960cab84fe2338f230b5e7535bc8efa9f9778ecf61c9f527067351dca895R619,